### PR TITLE
Validate update configuration even if label of cluster is changed

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
Signed-off-by: Hanwen <hanwenli@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/update/test_update.py [hunk 1]
```diff
@@ -464,9 +484,12 @@ def test_update_awsbatch(region, pcluster_config_reader, clusters_factory, test_
...
 
-    # Update cluster with the same configuration, command should not result any error even if not using force update
-    cluster.config_file = str(init_config_file)
-    cluster.update(force=False)
+    # Update cluster with the same configuration. Command should detect there is no change
+    _assert_no_update(cluster.name, init_config_file)
+
+    # Update cluster with another cluster section, where the configurations are the same as default cluster section.
+    # Command should treat the two sections as the same and detect there is no change
+    _assert_no_update(cluster.name, init_config_file, cluster_section_label="default2")
```
Comment (ddeidda) [inline tests/integration-tests/tests/update/test_update.py:492]: I'd do this check in the unit tests; this integration test could probably be left as it was. Let's talk about it if you want.

**Review by ddeidda — COMMENTED** [review]
Good job! I added just two commments; let's discuss them if you want.

File: tests/integration-tests/tests/update/test_update.py (hunk 1 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/tests/update/test_update.py:492]: So are you saying we do not need any integration test for this task?

**Review by ddeidda — APPROVED** [review]
Let's merge it!

**Pull request merged**